config/samples/mongodb.com_v1_mongodbcommunity_tls_cr.yaml

```diff
@@ -12,10 +12,19 @@ spec:
       modes: ["SCRAM"]
     tls:
       enabled: true
+      optional: true
       certificateKeySecretRef:
         name: tls-secret-name
       caConfigMapRef:
         name: tls-ca-configmap-name
+  statefulSet:
+    spec:
+      template:
+        spec:
+          containers:
+          - name: mongodb-agent
+            useSslForAllConnections: true
+            sslRequireValidServerCertificates: false
```

Comment:
These settings doesn't seem right. It is the StatefulSet override, whereas settings are specific to the database. Depending on the expected outcome, you may want to have a look at the `automationConfig` to provide proper Automation Config override or `agent` to override Agent settings. 

For more context, see https://www.mongodb.com/docs/ops-manager/current/reference/cluster-configuration/#ssl